### PR TITLE
Add  `auc_u_test` to `performance_metrics.py`

### PR DESCRIPTION
**Pull request by bmacedo-lgtm**
**Reference Issues/PRs**
Fixes https://github.com/FunctionLab/selene/issues/70. Adds new AUC calculation. The previously used implementation is sklearn's `roc_auc_score`, which is currently offered as the default metric. 

**What does this implement/fix? Explain your changes.**
Offers a faster implementation of the AUC calculation.  Based off of the following: https://blog.revolutionanalytics.com/2017/03/auc-meets-u-stat.html

**What testing did you do to verify the changes in this PR?**
Compared `roc_auc_score`  to `auc_u_test` and confirmed that they calculate the same values up to the same precision. Also timed the methods and confirmed that this new implementation is about 10x faster.

### CODE REVIEW

**Comment by kathyxchen** [comment]
Hi @bmacedo-lgtm, thanks for this! :) 

Could you fill out the PR template (just copy and paste the questions from here: https://github.com/FunctionLab/selene/pull/126) so that we have that information consistently across the pull requests?

Also, you can add me as a reviewer to this PR

File: selene_sdk/utils/performance_metrics.py [hunk 1]
```diff
@@ -247,6 +247,30 @@ def get_feature_specific_scores(data, get_feature_from_index_fn):
     return feature_score_dict
 
 
+def auc_u_test(labels, predictions):
+
+    """ Outputs the area under the the ROC curve associated with a certain
```
Comment (kathyxchen) [inline selene_sdk/utils/performance_metrics.py]: remove space here between docstring `"""` and `Outputs`

Comment (kathyxchen) [reply selene_sdk/utils/performance_metrics.py]: Also specify in the description that this is the AUC computed from U statistic

File: selene_sdk/utils/performance_metrics.py [hunk 2]
```diff
@@ -247,6 +247,30 @@ def get_feature_specific_scores(data, get_feature_from_index_fn):
...
 
+def auc_u_test(labels, predictions):
+
+    """ Outputs the area under the the ROC curve associated with a certain 
+    set of labels and the predictions given by the training model.
+
+    Parameters
+    ----------
+    labels: numpy.ndarray
+    Known labels of values predicted by model.
```
Comment (kathyxchen) [inline selene_sdk/utils/performance_metrics.py]: Indent, specify the exact shape expected (i.e. is this 2 dimensional? what are the dimensions?)

File: selene_sdk/utils/performance_metrics.py [hunk 3]
```diff
@@ -247,6 +247,30 @@ def get_feature_specific_scores(data, get_feature_from_index_fn):
...
+
```
Comment (kathyxchen) [inline selene_sdk/utils/performance_metrics.py]: remove extra space here

File: selene_sdk/utils/performance_metrics.py [hunk 4]
```diff
@@ -247,6 +247,30 @@ def get_feature_specific_scores(data, get_feature_from_index_fn):
...
+    numpy.float64
+    AUC value of given label, prediction pairs  
+    """
+    len_pos = int(np.sum(labels))
+    len_neg = len(labels) - len_pos
+    rank_sum = np.sum(rankdata(predictions)[labels == 1])
+    u_value = rank_sum - (len_pos * (len_pos + 1)) / 2
+    auc = u_value / (len_pos * len_neg)
+    return auc
+
```
Comment (kathyxchen) [inline selene_sdk/utils/performance_metrics.py]: add another newline here

File: selene_sdk/utils/performance_metrics.py (hunk 2 above)
Comment (kathyxchen) [reply selene_sdk/utils/performance_metrics.py]: (add indents where needed in the rest of the params/return)

File: selene_sdk/utils/performance_metrics.py [hunk 5]
```diff
@@ -247,6 +247,30 @@ def get_feature_specific_scores(data, get_feature_from_index_fn):
...
+    Parameters
+    ----------
+    labels: numpy.ndarray
+    Known labels of values predicted by model.
+
+    predictions: numpy.ndarray
+    Value predicted by user model
+
+    Returns:
+    numpy.float64
```
Comment (kathyxchen) [inline selene_sdk/utils/performance_metrics.py]: this can just be a float, it's not numpy-specific

File: selene_sdk/utils/performance_metrics.py [hunk 6]
```diff
@@ -247,6 +247,30 @@ def get_feature_specific_scores(data, get_feature_from_index_fn):
...
+    """ Outputs the area under the the ROC curve associated with a certain 
+    set of labels and the predictions given by the training model.
+
+    Parameters
+    ----------
+    labels: numpy.ndarray
+    Known labels of values predicted by model.
+
+    predictions: numpy.ndarray
+    Value predicted by user model
```
Comment (kathyxchen) [inline selene_sdk/utils/performance_metrics.py]: again, specify the expected shape

File: selene_sdk/utils/performance_metrics.py [hunk 7]
```diff
@@ -247,6 +247,37 @@ def get_feature_specific_scores(data, get_feature_from_index_fn):
...
+        Known labels of values predicted by model.
+        Must be one dimensional.
+    predictions: numpy.ndarray
+        Value predicted by user model
+        Must be one dimensional, with matching
+        dimension to `labels`
+
+    Returns
+    -------
+    float64
```
Comment (kathyxchen) [inline selene_sdk/utils/performance_metrics.py]: you can just specify this as float!

File: selene_sdk/utils/performance_metrics.py [hunk 8]
```diff
@@ -247,6 +247,37 @@ def get_feature_specific_scores(data, get_feature_from_index_fn):
...
+    """
+    Outputs the area under the the ROC curve associated with a certain 
+    set of labels and the predictions given by the training model.
+    Computed from the U statistic.
+
+    Parameters
+    ----------
+    labels: numpy.ndarray
+        Known labels of values predicted by model.
+        Must be one dimensional.
```
Comment (kathyxchen) [inline selene_sdk/utils/performance_metrics.py]: For this and the other parameter documentation, keep everything on 1 line (only creating a new line if the line exceeds 80 characters)

File: selene_sdk/utils/performance_metrics.py [hunk 9]
```diff
@@ -11,7 +11,7 @@
 from sklearn.metrics import precision_recall_curve
 from sklearn.metrics import roc_auc_score
 from sklearn.metrics import roc_curve
-
+from scipy.stats import rankdata
```
Comment (kathyxchen) [inline selene_sdk/utils/performance_metrics.py]: double check that there are 2 spaces between the import and the `logger` line?

File: selene_sdk/utils/performance_metrics.py [hunk 10]
```diff
@@ -247,6 +247,37 @@ def get_feature_specific_scores(data, get_feature_from_index_fn):
     return feature_score_dict
 
 
+def auc_u_test(labels, predictions):
+
```
Comment (kathyxchen) [inline selene_sdk/utils/performance_metrics.py]: remove newline here
